File: week-1/problem-sets/cash.c
#include <cs50.h>
#include <stdio.h>

int get_change();

int main(void)
{
    int change = 0;
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    int change_owed = get_change();

    int remainder = change_owed % quarter;
    change += change_owed / quarter;
    if (remainder != 0)
    {
        change += remainder / dime;
        remainder = remainder % dime;

        if (remainder != 0)
        {
            change += remainder / nickel;
            remainder = remainder % nickel;

            if (remainder != 0)
            {
                change += remainder / penny;
                remainder = remainder % penny;
            }
        }
    }
    printf("%i\n", change);
}

int get_change()
{
    int c;
    do
    {
        c = get_int("Change Owed: ");
    }
    while (c < 1);
    return c;
}

File: week-6/problem-sets/cash.py
def main():
    change = 0
    quarter = 25
    dime = 10
    nickel = 5
    penny = 1

    change_owed = get_change()
    remainder = change_owed % quarter
    change += change_owed // quarter
    if remainder != 0:
        change += remainder // dime
        remainder = remainder % dime
        if remainder != 0:
            change += remainder // nickel
            remainder = remainder % nickel
            if remainder != 0:
                change += remainder // penny
                remainder = remainder % penny
    print(int(change))


def get_change():
    while True:
        c = input("Change Owed: ")
        if c.replace(".", "", 1).isdigit():
            c = float(c)
            if c >= 0:
                return int(c * 100)


if __name__ == "__main__":
    main()
